public/components/common/debounced_component/debounced_component.test.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React from 'react';
+import { mountWithIntl as mount } from 'test_utils/enzyme_helpers';
+import { debouncedComponent } from './debounced_component';
+import { act } from 'react-dom/test-utils';
+
+describe('debouncedComponent', () => {
+  test('immediately renders', () => {
+    const TestComponent = debouncedComponent(({ title }: { title: string }) => {
+      return <h1>{title}</h1>;
+    });
+    expect(mount(<TestComponent title="hoi" />).html()).toMatchInlineSnapshot(`"<h1>hoi</h1>"`);
+  });
+
+  test('debounces changes', async () => {
+    const TestComponent = debouncedComponent(({ title }: { title: string }) => {
+      return <h1>{title}</h1>;
+    }, 1);
+    const component = mount(<TestComponent title="there" />);
+    component.setProps({ title: 'yall' });
+    expect(component.text()).toEqual('there');
+    await act(async () => {
+      await new Promise((r) => setTimeout(r, 1));
```

Comment:
not very familiar with the event loop, but would it be possible that this finishes before debounced component re-renders and test fails? since the delays are both `1`

Reply:
To me it looks like it would not fail since when setting the props, this action would be put into the event loop, then the following assertion statement would run immediately and pass. No matter what statement is actually picked by the main thread to run next (setting props to 'yall' or await act...), it would actually end up making sure the debounce component has changed its value to 'yall', before the last assertion runs. Since event handlers will be invoked in the same order that they were bound in.

Reply:
got it, looks good then